### DISCUSSION TITLE
hasOneDeep not working or I'm crazy

### DISCUSSION THREAD

**Discussion by thedangler** [staudenmeir/eloquent-has-many-deep#231] [Q&A]
I have the following:
Table A
subscription_id

Table Subscription
id
Order_Id

Table Order
id
Merchant_Id

Merchant
Id

How do I get the merchant using hasOneDeep from Table A

I tried $this->hasOneDeep(Merchant::class,[Order::class,Subscription::class])

but its not working. Its not joining the Table A.subscription_id on the subscription table.

**Comment by staudenmeir** [comment]
Hi @thedangler,
You need to provide all the foreign and local keys because of their custom structure:

```php
$this->hasOneDeep(
    Merchant::class,
    [Order::class,Subscription::class],
    ['id', 'id', 'id'],
    ['subscription_id', 'Order_Id', 'Merchant_Id']
)
```
